test/use_case/signup/SignupInteractorTest.java

```diff
@@ -0,0 +1,113 @@
+package use_case.signup;
+
+import data_access.InMemoryUsersDataAccessObject;
+import entity.Users.CommonUserFactory;
+import entity.Users.User;
+import entity.Users.UserFactory;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+
+public class SignupInteractorTest {
```

Comment:
The test coverage was just above 80%, I have added some tests to increase it to 94%.